### PR TITLE
fix Quick Info and Signature Help XML Doc indentation

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
image

image

### CODE REVIEW

**Comment by cartermp** [comment]
Very nice.

File: vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs [hunk 1]
```diff
@@ -40,11 +40,13 @@ type internal TextSanitizingCollector(collector, ?lineLimit: int) =
 
     let reportTextLines (s: string) =
         let mutable pos = 0
-        // skip newlines at the beginning
-        while pos < s.Length && isCROrLF s.[pos] do
+        // skip newlines and whitespaces at the beginning
+        while pos < s.Length && (Char.IsWhiteSpace s.[pos] || isCROrLF s.[pos]) do
```
Comment (KevinRansom) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]:
````
pos < s.Length && (Char.IsWhiteSpace s.[pos] 
````
Perhaps factor into a separate function it is used below ---- @line 64

Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]: I've removed the second copy of this line, so no need for a function anymore.

**Comment by vasily-kirichenko** [comment]
The way line breaks in XML Doc handled has been changed. 

RC2:

image

Current master:

image

This PR:

image

C#:

image

**Comment by vasily-kirichenko** [comment]
OK, I've rewritten `reportTextLines` in a more functional and much more safe way, now it looks like this:

image

image

image

**Comment by forki** [comment]
Also shorter ;-)

File: vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs [hunk 2]
```diff
@@ -23,49 +24,32 @@ type internal TextSanitizingCollector(collector, ?lineLimit: int) =
...
             collector (tagText "...")
             count <- count + 1
+        | _ ->
+            isEmpty <- false
+            endsWithLineBreak <- match text with TaggedText.LineBreak _ -> true | _ -> false
+            if endsWithLineBreak then count <- count + 1
+            collector text
     
-    let isCROrLF c = c = '\r' || c = '\n'
+    let splitTextRegex = Regex(@"\s*\n\s*\n\s*")
```
Comment (forki) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]: Should we add compiled option and make it static for perf?

**Comment by saul** [comment]
There seems to be a single space indent in the first screenshot before "Information about a symbol's definition" and in the third screenshot before "Mark a part as globally shared". 

I'm also not sure why only a method glyph is shown in the third screenshot when one of the symbols that it's showing is a class?

File: vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs [hunk 3]
```diff
@@ -385,8 +369,7 @@ module internal XmlDocumentation =
                 textCollector.Add(tagText errText)
                 true
 
-        List.fold Process false dataTipText
-        |> ignore
+        List.fold Process false dataTipText |> ignore
```
Comment (forki) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]: Fold + ignore? That's weird.

**Comment by vasily-kirichenko** [comment]
@saul Yes, the leading space presents. The problem with it that I cannot trim start of all lines, because they are concatenated around code blocks, line <text block><code block>< text block> (note the space in the last text block). I cannot see how to determine that the given text block is the first from XML Doc one.

About the glyph, we can provide only single one, FCS returns the symbol corresponding to the ctor, not the type, so we show "method" glyph.

File: vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs (hunk 3 above)
Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]: there are a lot of weird stuff there...

File: vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs (hunk 2 above)
Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs]: It's used to split 1-10 lines at a time, triggered by mouse move... It's not a place where we should think of any optimizations.

**Comment by vasily-kirichenko** [comment]
@saul I've fixed the heading space in XML Doc.

**Comment by vasily-kirichenko** [comment]
It's ready.

**Pull request merged**